demo/cypress/e2e/navigation.cy.js

```diff
@@ -1,55 +1,147 @@
+    beforeEach(() => {
+      loadComponentExample();
+      cy.viewport(800, 600);
+    });
+
+    it('has expected aria attributes', () => {
+      cy.get('.cads-greedy-nav').should('have.attr', 'aria-haspopup', 'true');
+      cy.findByRole('button', { name: /More/i }).should(
+        'have.attr',
+        'aria-controls',
+        'cads-greedy-nav-dropdown',
+      );
+    });
+
+    it('moves navigation items into the greedy navigation', () => {
+      cy.findByRole('button', { name: /More/i }).click();
+      assertItemsInMainNavigation([
+        'Benefits',
+        'Work',
+        'Debt and money',
+        'Consumer',
+        'Housing',
+        'Family',
+        'Law and courts',
+      ]);
+      assertItemsInGreedyNavigation(['Immigration', 'Health', 'More from us']);
+    });
   });
 
-  it('tabbing into the dropdown toggle automatically opens the dropdown menu', () => {
-    tabIntoNavigation();
-    assertNavigationOpen();
+  context('when on a small screen', () => {
+    beforeEach(() => {
+      loadComponentExample();
+      cy.viewport(375, 667);
+    });
+
+    it('moves navigation items into the greedy navigation including header links', () => {
+      cy.findByRole('button', { name: /More/i }).click();
+      assertItemsInMainNavigation(['Benefits', 'Work']);
+
+      assertItemsInGreedyNavigation([
+        'Debt and money',
+        'Consumer',
+        'Housing',
+        'Family',
+        'Law and courts',
+        'Immigration',
+        'Health',
+        'More from us',
+        'AdviserNet',
+        'Cymraeg',
+        'Sign in',
+      ]);
+    });
   });
 
-  // NP-1755
-  it.skip('tabbing out of the dropdown menu automatically closes it');
+  context('when interacting with a mouse', () => {
+    beforeEach(() => {
+      loadComponentExample();
+      cy.viewport(375, 667);
+    });
 
-  it('can close the dropdown menu using your keyboard', () => {
-    openNavigation().type('{esc}');
-    assertNavigationClosed();
+    it('allows opening and closing the greedy navigation', () => {
+      cy.findByRole('button', { name: /More/i }).click();
+      assertNavigationOpen();
+      cy.findByRole('button', { name: /Close/i }).click();
+      assertNavigationClosed();
+    });
+
+    it('closes the greedy navigation when clicking outside it', () => {
+      cy.findByRole('button', { name: /More/i }).click();
+      assertNavigationOpen();
+      cy.get('body').click();
+      assertNavigationClosed();
+    });
   });
 
-  it('can close the dropdown menu by clicking outside it', () => {
-    openNavigation();
-    assertNavigationOpen();
-    cy.get('body').click();
-    assertNavigationClosed();
+  // Note: Tab interaction is a bit fraught to test in Cypress
+  // We use https://github.com/dmtrKovalenko/cypress-real-events
+  // to approximate this but full keyboard testing should still be
+  // done manually when changing this behaviour.
+  context('when interacting with a keyboard', () => {
```

Comment:
Should we also test tabbing into it?  If it's too fraught at least some sort of note to test that behaviour manually too?

Reply:
I'll need to double check again, I think the answer is still that cypress doesn't handle that well so yes maybe a document that is manual testing steps for greedy nav would be good.